### PR TITLE
Feature createuser

### PR DESCRIPTION
**Pull request by plato-cambrian**
OK now you can merge me!

### CODE REVIEW

File: apps/chat/sopro.js [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+            return done(err);
+          }
+          opts.token = token;
+          done(null, opts);
+        });
+      },
+      sopro.crypto.saveToken,
+      function(opts, done){
+        // var html = "https://localhost/confirmUser/"+opts.token.token
+        // SES.sendEmail(opts.user.email, html, done)
```
Comment (voodoo-chile) [inline apps/chat/sopro.js]: extra comments?

File: apps/chat/tests/acceptance/features/step_definitions/postUserAPI_steps.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        });
+        f.on('message', function(msg, seqno) {
+          var prefix = '(#' + seqno + ') ';
+          msg.on('body', function(stream, info) {
+            var buffer = '';
+            stream.on('data', function(chunk) {
+              buffer += chunk.toString('utf8');
+            });
+            stream.once('end', function() {
+              //console.log(prefix + 'Parsed header: %s', inspect(Imap.parseHeader(buffer)));
```
Comment (voodoo-chile) [inline apps/chat/tests/acceptance/features/step_definitions/postUserAPI_steps.js]: debug comment

File: apps/chat/tests/acceptance/features/step_definitions/postUserAPI_steps.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
+var CAM_MOCKS = require('../../../mock-data.js');
+var S_STEPS = require('../../shared_steps.js');
+var SA_STEPS = require('../../sharedAPI_steps.js');
+var acl = require('../../../../auth-matrix.js')();
+var assert = require('assert');
+/*
+var PI = require('../../../../persistence-interface.js')();
+var PICouch = require('../../../../persistence-couchdb');
+PI.use(PICouch);
+*/
```
Comment (voodoo-chile) [inline apps/chat/tests/acceptance/features/step_definitions/postUserAPI_steps.js]: lose this comment block?

File: apps/chat/persistence-couchdb.js [hunk 4]
```diff
@@ -49,4 +49,17 @@ couch.destroy = function(data, callback){
   db.destroy(data._id, data._rev, callback);
 }
 
+couch.find = function(model, key, value, callback){
```
Comment (plato-cambrian) [inline apps/chat/persistence-couchdb.js]: needs unit tests

**Pull request merged**